### ISSUE TITLE
mapplot.plot - changing cube size to a small number does not zoom in on a region

### ISSUE THREAD

**Issue by KateRowlands** [sdss/marvin#327]
from marvin.tools.maps import Maps
import marvin.utils.plot.map as mapplot

maps = Maps(plateifu='8485-1901')
ha = maps['emline_gflux_ha_6564']
low_snr = mapplot.mask_low_snr(value=ha.value, ivar=ha.ivar, snr_min=10)

extent = mapplot.set_extent(cube_size=(1,1), sky_coords=True)

fig, ax = mapplot.plot(dapmap=ha, mask=low_snr, imshow_kws={'extent':extent})

Produces the following plot, which is the full size of the image but only the axis change, the plot does not zoom in on the central 1"x1" region as expected.

![screen shot 2018-02-20 at 11 22 15 am](https://user-images.githubusercontent.com/12090407/36435694-5a8d44e4-1630-11e8-9967-86cdb0fa0fe4.png)

**Comment by bretthandrews** [comment]
@KateRowlands thanks for the feedback! Try setting the axis range instead of the `extent` like this:

```
from marvin.tools.maps import Maps
import marvin.utils.plot.map as mapplot

maps = Maps(plateifu='8485-1901')
ha = maps['emline_gflux_ha_6564']
low_snr = mapplot.mask_low_snr(value=ha.value, ivar=ha.ivar, snr_min=10)

fig, ax = mapplot.plot(dapmap=ha, mask=low_snr, sky_coords=True)
ax.axis([-5, 5, -5, 5])
```

**Comment by KateRowlands** [comment]
Hi Brett,

Thanks for looking into this. Your suggestion works great, but does not involve the use of extent. Is the extent function going to be fixed, or is it redundant (and therefore should be removed from the documentation) since ax.axis job the same job?

Cheers

Kate

**Comment by bretthandrews** [comment]
Excellent point! After working through your use case, I realized that the extent should always be the full size of the map and that zooming in should be done via `ax.axis()`. `set_extent()` is working as it should, but the docs suggest that it should do what `ax.axis()` does (which is wrong). So this was really a docs problem (caused by my fundamental misunderstanding). That means that you do not need to use `set_extent`, and we should make it hidden to avoid confusion and remove it from the docs.

**Comment by KateRowlands** [comment]
OK thanks!

Is there any reason for extent to be in the code? If not then it should be completely removed and not hidden, in case someone finds it and uses it.

**Comment by bretthandrews** [comment]
`set_extent` toggles between sky and spaxel coordinates, so it shouldn't be removed completely, but it should be made for internal use only.

**Comment by havok2063** [comment]
Reopen.

**Comment by bretthandrews** [comment]
Fixed in 9222091. mapplot.set_patch_style and mapplot.set_extent have been converted into internal functions. There are examples in the Plotting Tutorial on how to create zoom in maps and how to change the hatch appearance (effectively what those functions had been intended to do).
